gridcomps/History/MAPL_HistoryCollection.F90

```diff
@@ -81,43 +93,43 @@ module MAPL_HistoryCollectionMod
      type(ESMF_FIELD), pointer          :: r8(:) => null()
      type(ESMF_FIELD), pointer          :: r4(:) => null()
      character(len=ESMF_MAXSTR)         :: output_grid_label
-     type(GriddedIOItemVector)            :: items
+     type(GriddedIOItemVector)          :: items
      character(len=ESMF_MAXSTR)         :: currentFile
      character(len=ESMF_MAXPATHLEN)     :: trackFile
      logical                            :: splitField
      logical                            :: regex
      logical                            :: timeseries_output = .false.
      logical                            :: recycle_track = .false.
      type(HistoryTrajectory)            :: trajectory
-     character(len=ESMF_MAXSTR)         :: positive 
+     character(len=ESMF_MAXSTR)         :: positive
+     type(HistoryCollectionGlobalAttributes) :: global_atts
      contains
         procedure :: AddGrid
-        procedure :: define_collection_attributes
   end type HistoryCollection
 
   contains
 
      function define_collection_attributes(this,rc) result(global_attributes)
-        class(HistoryCollection), intent(inout) :: this
+        class(HistoryCollectionGlobalAttributes), intent(inout) :: this
         integer, optional, intent(out) :: rc
 
         type(StringStringMap) :: global_attributes
         integer :: status
 
         call global_attributes%insert("Title",trim(this%descr))
         call global_attributes%insert("History","File written by MAPL_PFIO")
-        call global_attributes%insert("Source","unknown")
-        call global_attributes%insert("Contact","http://gmao.gsfc.nasa.gov")
-        call global_attributes%insert("Convention","CF")
-        call global_attributes%insert("Institution","NASA Global Modeling and Assimilation Office")
-        call global_attributes%insert("References","see MAPL documentation")
+        call global_attributes%insert("Source",trim(this%source))
+        call global_attributes%insert("Contact",trim(this%contact))
+        call global_attributes%insert("Convention",trim(this%convention))
+        call global_attributes%insert("Institution",trim(this%institution))
+        call global_attributes%insert("References",trim(this%references))
```

Comment:
If we could do allocatable it would obviate the need for TRIM() here.  Sigh.  Technically even now the TRIM is not required, but sort of wasteful in the gFTL container to keep the bytes.